### PR TITLE
Add performance tests github workflows

### PR DESCRIPTION
**Pull request by mjfwebb**
# Description

This PR adds performance test workflows for PRs. It will run the performance script on the PR's target and then the PR's branch followed by commenting the results on the PR.

## Complexity

Complexity: Medium

### CODE REVIEW

File: .github/workflows/performance-tests-comment.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - completed
+
+jobs:
+  performance-tests-comment:
+    if: github.event.workflow_run.conclusion == 'success'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Download built documentation"
```
Comment (tbwiss) [inline .github/workflows/performance-tests-comment.yml:16]: Wdyt?

```suggestion
      - name: "Download performance report"
```

**Review by tbwiss — APPROVED** [review]
Nice work!

File: .github/workflows/performance-tests-comment.yml (hunk 1 above)
Comment (mjfwebb) [reply .github/workflows/performance-tests-comment.yml:16]: Nice catch, thanks!

**Review by darrellwarde — APPROVED** [review]
Looks good, I have a feeling this should work!

**Pull request merged**